### PR TITLE
Ensure cargo-kani setup is idempotent

### LINKED ISSUE #1156: `cargo-kani setup` should be idempotent
Example counterexample. If cargo-kani setup fails after step 3 then subsequent calls to `cargo-kani setup` will fail at step 3 with:

```bash
bash-3.2$ cargo-kani setup
[0/6] Running Kani first-time setup...
[1/6] Ensuring the existence of: /Users/ncchong/.kani
[2/6] Downloading Kani release bundle: kani-0.1.0-x86_64-apple-darwin.tar.gz
[3/6] Installing rust toolchain version: nightly-2022-03-23-x86_64-apple-darwin
info: syncing channel updates for 'nightly-2022-03-23-x86_64-apple-darwin'

  nightly-2022-03-23-x86_64-apple-darwin unchanged - rustc 1.61.0-nightly (5f3700105 2022-03-22)

info: checking for self-updates
ln: ~/.kani/kani-0.1.0/toolchain/nightly-2022-03-23-x86_64-apple-darwin: File exists
Error: Failed command: ln -s ~/.rustup/toolchains/nightly-2022-03-23-x86_64-apple-darwin ~/.kani/kani-0.1.0/toolchain
```

A workaround for now is to remove the file manually:

```
$ rm ~/.kani/kani-0.1.0/toolchain/nightly-2022-03-23-x86_64-apple-darwin
```

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

Quick fixes for lack of idempotency in `cargo-kani setup`

### Resolved issues:

Resolves #1156

### Call-outs:

1. First fix is symlink creation. It would fail if it already existed. I also switched to using the native rust instead of shelling out to `ln -s`
2. Second fix was the rename in the `--use-local-bundle` code. I changed this to directly extract the bundle to the correct location in the first place, making its effective behavior more like the download case.
3. Minor fix: `canonicalize` can fail if the file doesn't exist. Print the file before canonicalizing.

### Testing:

* How is this change tested? A re-install test added to the release bundle tests

* Is this a refactor change? no

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by zhassan-aws — APPROVED** [review]
Needs a formatting fix, but otherwise looks good.

File: .github/workflows/kani.yml [hunk 1]
```diff
@@ -107,6 +107,7 @@ jobs:
           docker run -w /tmp/kani/tests/cargo-kani/simple-lib  kani-latest  cargo kani
           docker run -w /tmp/kani/tests/cargo-kani/simple-visualize  kani-latest  cargo kani
           docker run -w /tmp/kani/tests/cargo-kani/build-rs-works  kani-latest  cargo kani
+          docker run kani-latest  cargo-kani setup --use-local-bundle ./kani-latest-x86_64-unknown-linux-gnu.tar.gz
```
Comment (adpaco-aws) [inline .github/workflows/kani.yml:110]:
```suggestion
          docker run kani-latest cargo-kani setup --use-local-bundle ./kani-latest-x86_64-unknown-linux-gnu.tar.gz
```

**Review by adpaco-aws — COMMENTED** [review]
The first docker command in this group also contains a double whitespace. Can you please correct that too?

File: .github/workflows/kani.yml (hunk 1 above)
Comment (tedinski) [reply .github/workflows/kani.yml:110]: They all do, I was trying to keep the "command run in the container" more separate.

**Pull request merged**